src/main/java/edu/harvard/iq/dataverse/workflow/WorkflowServiceBean.java

```diff
@@ -171,10 +178,20 @@ private void doResume(PendingWorkflowInvocation pending, String body) {
         final WorkflowContext ctxt = refresh(newCtxt,retrieveRequestedSettings( wf.getRequiredSettings()), getCurrentApiToken(newCtxt.getRequest().getAuthenticatedUser()));
         WorkflowStepResult res = pendingStep.resume(ctxt, pending.getLocalData(), body);
         if (res instanceof Failure) {
+            userNotificationService.sendNotification(ctxt.getRequest().getAuthenticatedUser(), Timestamp.from(Instant.now()), UserNotification.Type.WORKFLOW_FAILURE, ctxt.getDataset().getLatestVersion().getId(), ((Failure) res).getMessage());
+            //UserNotification isn't meant to be a long-term record and doesn't store the comment, so we'll also keep it as a workflow comment
```

Comment:
Do I understand this correctly, that these notifications for workflow success and failure are email-only; in that you are purposefully excluding them from being shown on the dataverseuser.xhtml page? In part, I'm assuming, because of the above, that the custom comment/message doesn't get stored in the database. And with the assumption that the message will be shown to the user using the workflow UI messaging...

Reply:
... having read the PR description one more time though, you are saying
> a new enum type of UserNotification (resulting in an email *and addition on the notification tab*)

so, is it "purposefully", or not?

Reply:
The notification should be standard and show up in both places. The comment is just noting that UserNotification doesn't include a comment field, so the [comment goes out in the email but isn't saved as part of the UserNotification ](https://github.com/IQSS/dataverse/blob/bae37ca1ca086602a643a1493f86ad3a3b8c920d/src/main/java/edu/harvard/iq/dataverse/UserNotificationServiceBean.java#L105-L121). So, when I need to show something on the page which includes the comment, I needed a place to store the comment. 
An alternate design would be to add a comment field to UserNotification but that was bigger, not necessarily something we'd want for other notifications, and the WorkflowComment class seemed to be ~intended for this.

Reply:
Hmm. I may be being slow here, but I just didn't see any code being added to dataverseuser.xhtml, that would show these new notifications in that tab. So my first assumption was that it was on purpose. Although, on a 2nd or 3rd thought, that could be confusing - if the little red alert in the account header is showing that there's one new notification, but nothing new is showing in the notifications tab - ?

Reply:
Not intentional - I missed it, so the count goes up and a line in the table shows the date (done for all notifications), but there's not dataverseuser.xhtml fragment to provide the info specific to these types. I'll add them.

Reply:
OK, makes sense. I obviously didn't remember how that tab was working either; my recollection was that every table entry had to be hard-coded in full.